.github/workflows/ruby.yml

```diff
@@ -1,19 +1,42 @@
+---
 name: Ruby
 
-on: [push, pull_request]
+on:
+  pull_request: {}
+  push:
+    branches: main
 
 jobs:
-  build:
-
-    runs-on: ubuntu-latest
-
+  rubocop_and_matrix:
+    runs-on: ubuntu-24.04
+    outputs:
+      ruby: ${{ steps.ruby.outputs.versions }}
     steps:
     - uses: actions/checkout@v4
-    - name: Set up Ruby 2.7
+    - name: Set up Ruby
       uses: ruby/setup-ruby@v1
       with:
-        ruby-version: '2.7'
-        bundler: '2.4.22'
+        ruby-version: '3.4'
         bundler-cache: true
-    - name: Build and test with Rake
-      run: bundle exec rake
+    - id: ruby
+      uses: voxpupuli/ruby-version@v1
```

Comment:
this reads the ruby version requirement in the gemspec and returns a list of all major ruby versions that are allowed